### DISCUSSION TITLE
image tag in sources.yaml (local path)

### DISCUSSION THREAD

**Discussion by bvoightlab** [greenelab/lab-website-template#242] [General] [answered]
Hi there -

I wanted to try to take more direct use of the image tag in sources.yaml

where rather than specific an external link as a thumbnail

I instead refer to a local path which contains an image that I'd use

e.g., functionality that works with projects.yaml:

`image: images/mydirectory/myimage.png`

this would be useful if, say, we created cover art but then didn't get the cover but I still would like to use the representative image and uploaded this (or another thumbnail on the repo)

I tried to do that the intuitive way within sources.yaml (i.e., the above) but that didn't work

I did get the *substantially uglier* version of this to work by using the external link hosted at the website github page directly, 

e.g.

`image: https://github.com/bvoightlab/bvoightlab.github.io/blob/e34920f1f98cf3c1c0907822820d8d51ac93d407/images/covers/PMID_38374256_Cover_artonly.jpg?raw=true`

that works (see https://voightlab.com/publications/, citation "Genetic drivers of heterogeneity in type 2 diabetes pathophysiology) but .... you know. Brutal on the back end!

Thought maybe I was using the wrong syntax though, so posting for insight.

Best,
-BV

**Comment by vincerubinetti** [comment]
This should actually work, this was always intended functionality in the template, so something is wrong here. I'm unable to reproduce the error locally. Are you sure the image path was correct and that the image existed? And that the `image:` field started with `images` and not `./images` or `/images`. Can you point me to a place in your repo that is happening? Say, push a test update that includes an `image: images/mydirectory/myimage.png` so I can see what happens

**Comment by bvoightlab** [comment]
yeah sure thing

In my repo: 

`https://github.com/bvoightlab/bvoightlab.github.io/tree/main`

you can check out my `sources.yaml`: 

`https://github.com/bvoightlab/bvoightlab.github.io/blob/main/_data/sources.yaml`

As a mock up, you can see that very first reference/citation:
"Major Depressive Disorder Impacts Peripheral Artery Disease Risk Through Intermediary Risk Factors"

includes the tag

`image: images/covers/PMID_38374256_Cover_artonly.jpg`

I copied the path directly from the github site which is here

https://github.com/bvoightlab/bvoightlab.github.io/tree/main/images/covers

so I think the file is on the repo, the pathway is correct

but if you head to:

https://voightlab.com/publications/

and scroll down, you don't see the thumbnail attached to that very first citation
as a control, the second citation does spell out the link to the image on the github repo and will display the thumbnail

so there's something non-obvious to me that I'm not doing right that maybe you can see!

Best,
-BV

**Reply by vincerubinetti** [reply]
This is very strange. I was able to get this working locally, copying your exact `sources.yaml` entries and `images/cover` image, but with an otherwise "fresh" version of the template:

<img width="1111" alt="Screenshot 2024-02-27 at 2 00 30 PM" src="https://github.com/greenelab/lab-website-template/assets/8326331/c47ae824-8df7-4d64-94c4-328889bb363c">

Looking at the raw html generated at `view-source:https://voightlab.com/publications/`, line 751, `src=""` is empty. https://github.com/bvoightlab/bvoightlab.github.io/blob/gh-pages/publications/index.html#L730 This would come directly from `_includes/citation.html` `src="{{ citation.image | relative_url }}"`.

The image itself is also being uploaded and hosted just fine: https://voightlab.com/images/covers/PMID_38374256_Cover_artonly.jpg 


I will try to clone your repo later today to investigate more, but if you want to debug more in the meantime, you could add `{{ citation | inspect }}` above `<div class="citation">` and see what it prints.

**Reply by vincerubinetti** [reply]
Note that while investigating this, I did find another bug, but I think it's unrelated. The fix is here if you want to try applying it:
https://github.com/greenelab/lab-website-template/commit/f9a9d7bc0d734238622819fbc6efc9a3d69ffef1

But I don't see how that could be causing the problem in your case... it should only happen if you had a manually entered source without an id, and you don't. And if it did happen, the whole citation would be wrong/a duplicate of another citation, but in your case the citation seems to show up correctly except for the image.

**Reply by vincerubinetti** [reply, marked as answer]
Well I have bad news... I cloned your repo exactly how it is, and could not replicate the error you're seeing. I even pushed it to github and let it build on github actions:

https://vincerubinetti.github.io/publications#2024
https://github.com/vincerubinetti/vincerubinetti.github.io/actions

So, exact same code, exact same runtime environments, yet somehow in your repo a `src` is mysteriously missing. Maybe try clearing your [ruby cache on github actions](https://github.com/bvoightlab/bvoightlab.github.io/actions/caches)? 

In the `build-site.yaml` workflow, you could insert this debugging step somewhere and ssh right into the github actions environment while it's running, and try to determine why that `src` is blank when building... 

```yaml
- uses: mxschmitt/action-tmate@v3
```

I'll have v1.2.0 of the template out this week, and perhaps upgrading everything will solve it.
